### PR TITLE
Create model api

### PR DESCRIPTION
**Pull request by samihamdan**
One way to implement #114

### CODE REVIEW

File: julearn/api.py [hunk 1]
```diff
@@ -216,3 +212,29 @@ def run_cross_validation(
...
+
+
+def create_pipeline(
+    model,
+    preprocess_X=None,
+    preprocess_y=None,
+    preprocess_confounds=None,
+    confounds=None,
+    problem_type='binary_classification',
+):
```
Comment (fraimondo) [inline julearn/api.py:224]: Missing docstring

**Review by fraimondo — CHANGES_REQUESTED** [review]

**Comment by samihamdan** [comment]
Added functionality to set model/pipeline params. 
No custom checks on it, because it will throw the proper error anyways if someone tries to set smth wrong

File: julearn/api.py [hunk 2]
```diff
@@ -218,3 +215,86 @@ def run_cross_validation(
...
+        See documentation for details.
+    preprocess_confounds : str, scikit-learn transformers or list | None
+        Transformer to apply to the features (X). If string, use one of the
+        available transformers. If list, each element can be a string or
+        scikit-learn compatible transformer. If None (default), no
+        transformation is applied.
+
+        See documentation for details.
+    model_params : dict | None
+        If not None it has to be a dictionary of paramaeter value pair.
```
Comment (fraimondo) [inline julearn/api.py:273]: typo

File: julearn/api.py [hunk 3]
```diff
@@ -218,3 +215,86 @@ def run_cross_validation(
...
+    preprocess_X, preprocess_y, preprocess_confounds = preprocess_vars
+    # Prepare the model
+    model_tuple = prepare_model(model=model, problem_type=problem_type)
+
+    pipeline = _create_extended_pipeline(preprocess_X,
+                                         preprocess_y,
+                                         preprocess_confounds,
+                                         model_tuple, confounds,
+                                         categorical_features=None)
+    if model_params is not None:
```
Comment (fraimondo) [inline julearn/api.py:297]: `prepare_model_params`

**Review by fraimondo — CHANGES_REQUESTED** [review]

File: docs/changes/latest.inc [hunk 4]
```diff
@@ -24,6 +24,9 @@ Enhancements
 - Refactor scoring to allow for registering and callable scorers (by `Sami Hamdan`_).
 
 - Update :mod:`.model_selection` and add capabilities to register searchers (by `Sami Hamdan`_).
+  - Add user facing `create_pipeline` function (by `Sami Hamdan`_)
```
Comment (fraimondo) [inline docs/changes/latest.inc:27]: Extra space, missing newline

**Review by fraimondo — CHANGES_REQUESTED** [review]

**Review by fraimondo — APPROVED** [review]

**Pull request merged**